tests/unit/test_charm.py

```diff
@@ -431,14 +423,123 @@ def test_given_unit_certificate_not_stored_when_configure_then_unit_certificate_
 
         self.harness.charm.on.config_changed.emit()
 
-        patch_generate_unit_certificate.assert_called_with(
-            subject=ingress_address,
-            sans_ip=[ingress_address],
-            sans_dns=[fqdn],
-            ca_certificate=ca_certificate.encode(),
-            ca_private_key=ca_private_key.encode(),
+        assert (
+            (root / "vault/certs/cert.pem").read_text().startswith("-----BEGIN CERTIFICATE-----")
+        )
+
+    def test_given_certificate_access_relation_when_relation_joined_then_new_request_is_created(
+        self,
+    ):
+        self.harness.set_leader(is_leader=True)
+        self.harness.add_storage(storage_name="certs", attach=True)
+        root = self.harness.get_filesystem_root(self.container_name)
+        self.harness.set_can_connect(container=self.container_name, val=True)
+        (root / "vault/certs/key.pem").write_text(EXAMPLE_PK)
+
+        self._set_peer_relation()
+        self._set_tls_access_certificate_relation()
+        self.harness.charm.tls.configure_certificates("1.1.1.1")
```

Comment:
Tests for the new tls lib should go in their own file

Reply:
Will do

Reply:
Done